client/js/pages/chatterbox.js

```diff
@@ -244,12 +244,155 @@ export default function ($parent, options) {
 			}
 		};
 
+		function ButtsFactory() {
+			this.newButts = function (buttFrequency, jumbleFrequency){
+				var butts = {};
+				butts.buttFrequency = buttFrequency;
+				butts.jumbleFrequency = jumbleFrequency;
+				butts.buttArray = [];
+				
+				butts.init = function(initArray){
+					butts.buttArray = initArray.slice(0);
+				}
+				
+				butts.buttTime = function(){
+					if(Math.floor(Math.random() * butts.buttFrequency) == 0){
+						return true;
+					}
+				}
+				
+				butts.buttify = function (word){ 
+					var result = word;
+					if(this.buttTime()){
+						if(Math.floor(Math.random() * 2) == 0){
+							result = "butt";
+						} else {
+							result = "butts";
+						}
+					}
+					if(school){
+						result = cap(result);
+					}
+					return result;
+				}
+				
+				butts.big = function (word){
+					if(Math.floor(Math.random() * 15) == 0){
+						return word.toUpperCase();
+					} else {
+						return word;
+					}
+				}
+				
+				butts.cap = function (string) {
+					return string.charAt(0).toUpperCase() + string.slice(1);
+				}
+				
+				butts.pick = function (a){
+					var i = Math.floor(a.length * Math.random());
+					return a[i];
+				}
+				
+				butts.emptyKiller = function(){
+					butts.buttArray = butts.buttArray.filter(butt => butt.trim() != '');
+					return this;
```

Comment:
I have an open TODO to support ES6 (with uglify), in the meanwhile change this to use a function, instead of arrow operator (=>)

filter(function(butt) { ...})...